### PR TITLE
[Robust control] Add makeweight function

### PR DESCRIPTION
**Pull request by don4get**
_To be merged after https://github.com/python-control/python-control/pull/270_

I wanted to add this functionality to the library:
https://ch.mathworks.com/help/robust/ref/makeweight.html

Above the Matlab behavior, I added the possibility to compute higher orders weight functions.

The discrete behavior is not implemented yet as I don't know how to deal with it.
I tried the naive approach, creating the laplace variable as follows: s = tf([1, 0], [1], Ts).
However, the DC gain of the resulted weight function is not correct, as one could have expected.
If someone knows how to solve this problem, I would be glad to help him.

### CODE REVIEW

File: control/robust.py [hunk 1]
```diff
@@ -136,26 +141,26 @@ def hinfsyn(P,nmeas,ncon):
...
-    #Check for ss system object, need a utility for this?
+    # Check for ss system object, need a utility for this?
 
-    #TODO: Check for continous or discrete, only continuous supported right now
-        # if isCont():
-        #    dico = 'C'
-        # elif isDisc():
-        #    dico = 'D'
-        # else:
+    # TODO: Check for continous or discrete, only continuous supported right now
```
Comment (murrayrm) [inline control/robust.py]: We should raise an exception if a discrete time system is passed as an argument.  Use `isdtime(P, strict=True)` to check.

File: control/robust.py [hunk 2]
```diff
@@ -356,8 +362,60 @@ def mixsyn(g,w1=None,w2=None,w3=None):
...
+    ----------
+        dcgain : double positive scalar
+            Low frequency gain of 1/W; should be < 1
+        wc : double strictly positive scalar
+            Design frequency (where |W| is approximately 1)
+        hfgain : double positive scalar
+            High frequency gain of 1/W; should be > 1
+        order : int strictly positive scalar - optional
+            Order of the output transfer function
+        Ts : double strictly positive scalar - optional
```
Comment (murrayrm) [inline control/robust.py]: Shouldn't the sampling time come from the system (`sys.dt`)?  What happens if they are not compatible?

File: control/robust.py [hunk 3]
```diff
@@ -356,8 +362,60 @@ def mixsyn(g,w1=None,w2=None,w3=None):
...
 
-    k,cl,gamma,rcond=hinfsyn(p,nmeas,ncon)
-    info = gamma,rcond
-    return k,cl,info
+def makeweight(dcgain, wc, hfgain, order=1, Ts=None):
+    """ Compute a weight transfer function, noted W
+
+    Parameters
+    ----------
+        dcgain : double positive scalar
```
Comment (murrayrm) [inline control/robust.py]: Incorrect indentation; should line up with `Parameters`.  See [numpydocs](http://numpydoc.readthedocs.io).

File: control/robust.py [hunk 4]
```diff
@@ -356,8 +362,60 @@ def mixsyn(g,w1=None,w2=None,w3=None):
...
+            Low frequency gain of 1/W; should be < 1
+        wc : double strictly positive scalar
+            Design frequency (where |W| is approximately 1)
+        hfgain : double positive scalar
+            High frequency gain of 1/W; should be > 1
+        order : int strictly positive scalar - optional
+            Order of the output transfer function
+        Ts : double strictly positive scalar - optional
+            Sampling time in case W is discrete
+    W - SISO LTI object
```
Comment (murrayrm) [inline control/robust.py]: Should be in a section titled `Returns`.  See [numpydoc](http://numpydoc.readthedocs.io).

File: control/robust.py [hunk 5]
```diff
@@ -356,8 +362,60 @@ def mixsyn(g,w1=None,w2=None,w3=None):
...
+    info = gamma, rcond
+    return k, cl, info
+
 
-    k,cl,gamma,rcond=hinfsyn(p,nmeas,ncon)
-    info = gamma,rcond
-    return k,cl,info
+def makeweight(dcgain, wc, hfgain, order=1, Ts=None):
+    """ Compute a weight transfer function, noted W
+
```
Comment (murrayrm) [inline control/robust.py]: Need more documentation about the functionality that is implemented.  Add a few sentences (and maybe an equation?) describing what the function does.

Also, this function needs to be added to `doc/control.rst` and `doc/matlab.rst`.

**Comment by iljastas** [comment]
is the weighting function of the robust_mimo.py example implemented also directly in the module?

**Comment by murrayrm** [comment]
PR has been idle for > 6  months => stale.  @don4get I will close this PR in a few days unless you will have time to work on it soon.  No worries if not; just re-open when you have responded to the comments above.

**Pull request closed without merging**